tic/tic.go

```diff
@@ -0,0 +1,1330 @@
+// information, see the Tic user's guide.
+func (d *Dev) ClearDriverError() error {
+	return d.commandQuick(cmdClearDriverError)
+}
+
+// GetMaxSpeed gets the current maximum speed, in microsteps per 10000 seconds.
+//
+// This is the current value, which could differ from the value in the Tic's
+// settings.
+func (d *Dev) GetMaxSpeed() (uint32, error) {
+	return d.getVar32(OffsetSpeedMax)
+}
+
+// SetMaxSpeed sets the maximum speed, in units of steps per 10000 seconds.
+//
+// Example:
+//
+//	err := dev.SetMaxSpeed(5550000) // 555 steps per second
+//
+// This function sends a "Set max speed" command to the Tic. For more
+// information, see the Tic user's guide.
+func (d *Dev) SetMaxSpeed(speed uint32) error {
+	return d.commandW32(cmdSetSpeedMax, speed)
+}
+
+// GetStartingSpeed gets the starting speed in microsteps per 10000 seconds.
+//
+// This is the current value, which could differ from the value in the Tic's
+// settings.
+func (d *Dev) GetStartingSpeed() (uint32, error) {
+	return d.getVar32(OffsetStartingSpeed)
+}
+
+// SetStartingSpeed sets the starting speed, in units of steps per 10000
+// seconds.
+//
+// Example:
+//
+//	err := dev.SetStartingSpeed(500000) // 50 steps per second
+//
+// This function sends a "Set starting speed" command to the Tic. For more
+// information, see the Tic user's guide.
+func (d *Dev) SetStartingSpeed(speed uint32) error {
+	return d.commandW32(cmdSetStartingSpeed, speed)
+}
+
+// GetMaxAccel gets the maximum acceleration, in microsteps per second per 100
+// seconds.
+//
+// This is the current value, which could differ from the value in the Tic's
+// settings.
+func (d *Dev) GetMaxAccel() (uint32, error) {
+	return d.getVar32(OffsetAccelMax)
+}
+
+// SetMaxAccel sets the maximum acceleration, in units of steps per second per
+// 100 seconds.
+//
+// Example:
+//
+//	err := dev.SetMaxAccel(10000) // 100 steps per second per second
+//
+// This function sends a "Set max acceleration" command to the Tic. For more
+// information, see the Tic user's guide.
+func (d *Dev) SetMaxAccel(accel uint32) error {
+	return d.commandW32(cmdSetAccelMax, accel)
+}
+
+// GetMaxDecel gets the maximum deceleration, in microsteps per second per 100
+// seconds.
+//
+// This is the current value, which could differ from the value in the Tic's
+// settings.
+func (d *Dev) GetMaxDecel() (uint32, error) {
+	return d.getVar32(OffsetDecelMax)
+}
+
+// SetMaxDecel sets the maximum deceleration, in units of steps per second per
+// 100 seconds.
+//
+// Example:
+//
+//	err := dev.SetMaxDecel(10000) // 100 steps per second per second
+//
+// This function sends a "Set max deceleration" command to the Tic. For more
+// information, see the Tic user's guide.
+func (d *Dev) SetMaxDecel(decel uint32) error {
+	return d.commandW32(cmdSetDecelMax, decel)
+}
+
+// StepMode describes how many microsteps add up to one fulls step.
+type StepMode uint8
+
+const (
+	StepModeFull StepMode = 0
+	StepModeHalf StepMode = 1
```

Comment:
This looks strange that StepModeHalf == StepModeMicrostep2 ? I don't see these two values (including StepModeFull) being used anywhere.